docs/AppFramework.md

```diff
@@ -0,0 +1,315 @@
+   * The apps are split across 3 folders named `networkApps`, `clusterBase`, and `adminApps` . The apps placed into  `networkApps` and `clusterBase` are distributed to the cluster peers, but the apps in `adminApps` are for local use on the cluster master instance only.
+
+4. Copy your Splunk App or Add-on archive files to the App Source.
+   * In this example, the Splunk Apps for the cluster peers are located at `bucket-app-framework-us-west-2/Standalone-us/networkAppsLoc/`,  `bucket-app-framework-us-west-2/Standalone-us/clusterBaseLoc/`, and the apps for the cluster master are located at`bucket-app-framework-us-west-2/Standalone-us/adminAppsLoc/`. They are all accessible through the end point `https://s3-us-west-2.amazonaws.com`.
+
+5. Update the ClusterMaster CR specification and append the volume, App Source configuration, and scope.
+   * The scope determines where the apps and add-ons are placed into the Splunk Enterprise instance. For CR's where the Splunk Enterprise instance will deploy the apps to cluster peers, set the `scope:  cluster`. The ClusterMaster and SearchHeadCluster CR's support both cluster and local scopes.
+   * In this example, the cluster master will run some apps locally, and deploy other apps to the cluster peers. The App Source folder `adminApps` are Splunk Apps that are installed on the cluster master, and will use a local scope. The apps in the App Source folders `networkApps` and `clusterBase` will be deployed from the cluster master to the peers, and will use a cluster scope.
+
+Example: ClusterMaster.yaml
+
+```apiVersion: enterprise.splunk.com/v1
+kind: ClusterMaster
+metadata:
+  name: cm
+  finalizers:
+  - enterprise.splunk.com/delete-pvc
+spec:
+  replicas: 1
+  appRepo:
+    appsRepoPollIntervalSeconds: 900
+    defaults:
+      volumeName: volume_app_repo_us
+      scope: cluster
+    appSources:
+      - name: networkApps
+        location: networkAppsLoc/
+      - name: clusterBase
+        location: clusterBaseLoc/
+      - name: adminApps
+        location: adminAppsLoc/
+        scope: local
+    volumes:
+      - name: volume_app_repo_us
+        storageType: s3
+        provider: aws
+        path: bucket-app-framework-us-west-2/Standalone-us/
+        endpoint: https://s3-us-west-2.amazonaws.com
+        secretRef: s3-secret
+```
+
+6. Apply the Customer Resource specification: `kubectl -f apply ClusterMaster.yaml`
+
+
+### How to use the App Framework on Search Head Cluster
+
+This example, describes installation of apps on Search Head Cluster as well as Deployer. This is achieved by deploying a SearchHeadCluster CR with a storage volume, the location of the app archives, and set the installation scope to support both local and cluster app distribution.
+
+1. Confirm your S3-based storage volume path and URL.
+2. Create a Kubernetes Secret Object with the storage credentials. 
+   * Example: `kubectl create secret generic s3-secret --from-literal=s3_access_key=$7$FLa0GfJNEzdcqLBcZmW1SDg==` 
```

Comment:
We verified that we only need the access key, correct?

Reply:
Seems like I jumped on this early. Our code fails without this. May be we need to give a try, and understand it more.